needle-zendesk-connector/src/server/api/routers/connectors-router.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  ConnectorRequestSchema,
+  CreateConnectorRequestSchema,
+  ZendeskResponseSchema,
+} from "~/models/connectors-models";
+import { createTRPCRouter, procedure } from "~/server/api/trpc";
+import {
+  createZendeskConnector,
+  deleteZendeskConnector,
+  getZendeskConnector,
+  listZendeskConnectors,
+  runZendeskConnector,
+} from "~/server/backend/connectors-backend";
+import { createZendeskService } from "~/server/zendesk/service";
+
+export const connectorsRouter = createTRPCRouter({
+  create: procedure
+    .input(CreateConnectorRequestSchema)
+    .mutation(async ({ ctx, input }) =>
+      createZendeskConnector(input, ctx.session),
+    ),
+
+  list: procedure.query(async ({ ctx }) => listZendeskConnectors(ctx.session)),
+
+  get: procedure
+    .input(ConnectorRequestSchema)
+    .query(async ({ ctx, input }) => getZendeskConnector(input, ctx.session)),
+
+  delete: procedure
+    .input(ConnectorRequestSchema)
+    .mutation(async ({ ctx, input }) =>
+      deleteZendeskConnector(input, ctx.session),
+    ),
+
+  run: procedure
+    .input(ConnectorRequestSchema)
+    .mutation(async ({ ctx, input }) =>
+      runZendeskConnector(input, ctx.session),
+    ),
+
+  getData: procedure.input(ZendeskResponseSchema).query(async ({ input }) => {
```

Comment:
This request to zendesk and directly go to zendesk API from the client right? We don't need to proxy it over our backend. Can we change that? (i.e. using Zendesk Service directly in the frontend)

In the frontend then the usage would look something like:

```js
const { data: jiraClouds } = useSuspenseQuery({
    queryKey: ["zendesk-data", accessToken],
    queryFn: () => zendeskService.listTicketsAndArticles(accessToken),
  });
```

i.e. using `useQuery`, or `useSuspenseQuery` from tanstack package.

Reply:
Yeah good thinking, trpc is redundant, I will apply the changes